backend/src/data/repositories/course/course.repository.ts

```diff
@@ -24,6 +24,7 @@ class Course {
     title: string;
   }): Promise<CourseGetResponseDto[]> {
     const { categoryId, title } = filteringOpts ?? {};
+    const normalizedTitle = title.split('\\').join('\\\\');
```

Comment:
Why just not to use `.replace`?

Reply:
I've tried this and it doesn't work properly for cases when there is more then 1 backslash in the search request: 

> const normalizedTitle = title.replace('\\', '\\\\');

Reply:
`replaceAll('\\', '\\\\')` ?

Reply:
Wait, this escaped '\\' comes from querystring parser in controller. Shouldn't we fix the cause of the problem, not the consequences?

Reply:
What do you mean by the cause?

Reply:
Frontend send `\a`
On backend in controller we got `\\a` - node:querystring escaped backslash, I believe this is why search not working as intended, because it's incorrectly parsed in controller

Reply:
@PavelSukhinin  yes and no. It does adds backslash and it would work, but knex adds E to all string containing double backslash which leads to double escape and the fact that we need more backslashes. Hope this makes sense.

Reply:
Ok, I got it, thanks 👍